packages/hisat2/package.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import glob
+
+from spack.package import *
+
+
+class Hisat2(MakefilePackage):
+    """HISAT2 is a fast and sensitive alignment program for mapping
+    next-generation sequencing reads (whole-genome, transcriptome, and
+    exome sequencing data) against the general human population (as well as
+    against a single reference genome)."""
+
+    homepage = "https://daehwankimlab.github.io/hisat2/"
+    url = "ftp://ftp.ccb.jhu.edu/pub/infphilo/hisat2/downloads/hisat2-2.1.0-source.zip"
+
+    version(
+        "2.2.1",
+        sha256="48e933330d4d8470d2b3dfe7ec3918f2e98a75f7381891e23b7df1fb4f135eb1",
+        url="https://cloud.biohpc.swmed.edu/index.php/s/fE9QCsX3NH4QwBi/download",
+        extension="zip",
+    )
+    version(
+        "2.2.0",
+        sha256="0dd55168853b82c1b085f79ed793dd029db163773f52272d7eb51b3b5e4a4cdd",
+        url="https://cloud.biohpc.swmed.edu/index.php/s/hisat2-220-source/download",
+        extension="zip",
+    )
+    version("2.1.0", sha256="89a276eed1fc07414b1601947bc9466bdeb50e8f148ad42074186fe39a1ee781")
+
+    # variant("sra", default=False, description="Add SRA (Sequence Read Archive) support")
+
+    depends_on("perl", type="run")
+    depends_on("python", type="run")
+    depends_on("python@:2", when="@:2.2.0")
```

Comment:
It might be better to express it with a  `conflicts("^python@3:", when="@:2.2.0", msg="...")` 
Except if the explicitly said that `@2.2.1` does not work with `python@2`

Reply:
Their code changed (among other things from):
`#!/usr/bin/env python`
to:
`#!/usr/bin/env python3`

so I think they consider it python3 only :-D

Reply:
Upon further inspection, it's a bit of a mess.

On 2.2.1:
`hisat2-build` is compatible with both python2 and python3 (or so it seems)
`hisat2_extract_splice_sites.py` (any *.py it seems) is only python3.

I think this makes it python3-only, and the current structure is fine, but I'll wait for one final comment from you.